binlog_streamer.go

```diff
@@ -133,6 +142,90 @@ func (s *BinlogStreamer) ConnectBinlogStreamerToMysqlFrom(startFromBinlogPositio
 	return s.lastStreamedBinlogPosition, err
 }
 
+func (s *BinlogStreamer) defaultEventHandler(ev *replication.BinlogEvent, query []byte, es *BinlogEventState) ([]byte, error) {
```

Comment:
I didn't go through the function of this body very carefully, but I assume it is copied more or less verbatim from the original code (with some minor adjustments to the variable names).

Reply:
One thing that is a bit surprising (maybe not?) is that the `es` parameter is being modified, and the "other loop" in the binlog streamer code is relying on that being modified to work (the `isEventResumableVariable`). Thus, to understand how the binlog streamer works works, you have to go into this function and trace the evolution of the parameter `es`, which to me feels a bit more convoluted than before. THAT SAID, in the absence of a bigger refactor, I think this is fine, but may be worth pointing out somewhere in a comment.

Or maybe I'm writing too much C++ these days and the lack of the `const` qualifier is getting to me 😅. Let me know either way.